Filters/MetadataOptionsFilter.cs

```diff
@@ -0,0 +1,33 @@
+using AngleSharp.Css;
+using Etch.OrchardCore.Greenhouse.Models;
+using Etch.OrchardCore.Greenhouse.Utilities;
+using Fluid;
+using Fluid.Values;
+using OrchardCore.Liquid;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Etch.OrchardCore.Greenhouse.Filters
+{
+    public class MetadataOptionsFilter : ILiquidFilter
+    {
+        public async ValueTask<FluidValue> ProcessAsync(FluidValue input, FilterArguments arguments, LiquidTemplateContext context)
+        {
+            var projects = new List<string>();
+            var selectedItem = arguments.At(0).ToStringValue();
```

Comment:
This isn't right as it doesn't cater for the arguments being the currently selected item AND the name of the meta field property that should show.

Instead of fetching the first argument, you should use `arguments[propertyName`] and `arguments[`selectedItem`] to get the two different arguments. The filter will then looking like this...

```
{{ allPostings | greenhouse_metadata_options: property: "Project", selectedItem: project | raw }}
```